core/src/test/java/org/elasticsearch/cluster/serialization/DiffableTests.java

```diff
@@ -96,8 +97,81 @@ public TestDiffable readFrom(StreamInput in) throws IOException {
         assertThat(serialized.get("foo").value(), equalTo("1"));
         assertThat(serialized.get("baz").value(), equalTo("4"));
         assertThat(serialized.get("new").value(), equalTo("5"));
+    }
+
+    public void testImmutableOpenIntMapDiff() throws IOException {
+        ImmutableOpenIntMap.Builder<TestDiffable> builder = ImmutableOpenIntMap.builder();
```

Comment:
can we use more of randomization? create two completely random sets and try to move from one to another.. reproducible evil FTW